packages/shortcuts-extension/src/components/ShortcutItem.tsx

```diff
@@ -315,6 +351,106 @@ export class ShortcutItem extends React.Component<
     );
   }
 
+  // handle key down function to navigate within each row
+  handleKeyDown(event: React.KeyboardEvent): void {
```

Comment:
This entire function needs to be refactored:
- `tabindex` manipulations should be explained
- typing should be fixed
- repetitive code should be minimalised